alibi/models/pytorch/tests/test_model.py

```diff
@@ -0,0 +1,618 @@
+@pytest.mark.parametrize('multimodal_model', [
+    {
+        'input_dim': input_dim,
+        'output_dims': output_dims,
+        'compile': True,
+        'optimizer': optimizer_class,
+        'loss': loss_fns,
+        'loss_weights': loss_weights,
+    }
+], indirect=True)
+def test_reset_loss_multimodal(multimodal_model, dataset):
+    """ Test if the model is resetting the loss function in the multimodal case. """
+    y_pred = dataset['output']['multimodal']['y_pred']
+    y_true = dataset['output']['multimodal']['y_true']
+
+    for i in range(2):
+        multimodal_model.loss[i](y_true=y_true[i], y_pred=y_pred[i])
+
+    multimodal_model._reset_loss()
+
+    for i in range(2):
+        assert np.isclose(multimodal_model.loss[i].total, 0)
+        assert np.isclose(multimodal_model.loss[i].count, 0)
+
+
+@pytest.mark.parametrize('dataset', ['classification'], indirect=True)
+@pytest.mark.parametrize('unimodal_model', [
+    {
+        'input_dim': input_dim,
+        'output_dim': output_dim,
+        'compile': True,
+        'optimizer': optimizer_class,
+        'loss': loss_fn,
+        'metrics': metric,
+    }
+], indirect=True)
+def test_reset_metrics_unimodal(unimodal_model, dataset):
+    """ Test if the model is resetting the metrics in the unimodal case. """
+    y_pred = dataset['output']['unimodal']['y_pred']
+    y_true = dataset['output']['unimodal']['y_true']
+
+    unimodal_model.metrics[0].compute_metric(y_true=y_true, y_pred=y_pred)
+    unimodal_model._reset_metrics()
+    assert np.isclose(unimodal_model.metrics[0].total, 0)
+    assert np.isclose(unimodal_model.metrics[0].count, 0)
+
+
+@pytest.mark.parametrize('dataset', ['classification'], indirect=True)
+@pytest.mark.parametrize('multimodal_model', [
+    {
+        'input_dim': input_dim,
+        'output_dims': output_dims,
+        'compile': True,
+        'optimizer': optimizer_class,
+        'loss': loss_fn,
+        'loss_weights': loss_weights,
+        'metrics': metrics,
+    }
+], indirect=True)
+def test_reset_metrics_multimodal(multimodal_model, dataset):
+    """ Test if the model is resetting the metrics in the multimodal case. """
+    y_pred = dataset['output']['multimodal']['y_pred']
+    y_true = dataset['output']['multimodal']['y_true']
+
+    for i in range(2):
+        key = f'output_{i + 1}'
+        multimodal_model.metrics[key].compute_metric(y_true=y_true[i], y_pred=y_pred[i])
+
+    multimodal_model._reset_metrics()
+
+    for i in range(2):
+        key = f'output_{i + 1}'
+        assert np.isclose(multimodal_model.metrics[key].total, 0)
+        assert np.isclose(multimodal_model.metrics[key].count, 0)
+
+
+def test_saving():
+    """ Test saving functionality. """
+    model1 = UnimodalModel(input_dim=input_dim, output_dim=output_dim)
+    model2 = UnimodalModel(input_dim=input_dim, output_dim=output_dim)
+
+    with tempfile.TemporaryDirectory() as temp_dir:
```

Comment:
`pytest` actually ships with a `tmp_path` fixture which can be used, saving some code and imports :) https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html